Backup-restores panics

/area backup
/kind bug

**What happened**:
Backup-restore panics:

```
INFO[0269] backup-restore stops leading...               actor=backup-restore-server
INFO[0269] HTTP server closed gracefully.                actor=backup-restore-server
panic: close of closed channel

goroutine 1 [running]:
github.com/gardener/etcd-backup-restore/pkg/server.(*BackupRestoreServer).runServer.func3()
	/go/src/backup_forked/etcd-backup-restore/pkg/server/backuprestoreserver.go:277 +0x70
github.com/gardener/etcd-backup-restore/pkg/leaderelection.(*LeaderElector).Run(0x1400080bc70, {0x107be43f0, 0x1400080bea0})
	/go/src/backup_forked/etcd-backup-restore/pkg/leaderelection/leaderelection.go:94 +0x784
github.com/gardener/etcd-backup-restore/pkg/server.(*BackupRestoreServer).runServer(0x1400088e270, {0x107be43f0, 0x1400080bea0}, 0x14000181e60)
	/go/src/backup_forked/etcd-backup-restore/pkg/server/backuprestoreserver.go:337 +0x678
github.com/gardener/etcd-backup-restore/pkg/server.(*BackupRestoreServer).Run(0x1400088e270, {0x107be43f0, 0x1400080bea0})
	/go/src/backup_forked/etcd-backup-restore/pkg/server/backuprestoreserver.go:114 +0x3ac
github.com/gardener/etcd-backup-restore/cmd.(*serverOptions).run(0x1094f5fc0?, {0x107be43f0, 0x1400080bea0})
	/go/src/backup_forked/etcd-backup-restore/cmd/options.go:84 +0x54
github.com/gardener/etcd-backup-restore/cmd.NewServerCommand.func1(0x14000794900?, {0x106e9727d?, 0x4?, 0x106e9720d?})
	/go/src/backup_forked/etcd-backup-restore/cmd/server.go:43 +0x1b4
github.com/spf13/cobra.(*Command).execute(0x14000798c08, {0x140005daae0, 0x2, 0x2})
	/go/src/backup_forked/etcd-backup-restore/vendor/github.com/spf13/cobra/command.go:1019 +0x810
github.com/spf13/cobra.(*Command).ExecuteC(0x14000463b08)
	/go/src/backup_forked/etcd-backup-restore/vendor/github.com/spf13/cobra/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(0x107be43f0?)
	/go/src/backup_forked/etcd-backup-restore/vendor/github.com/spf13/cobra/command.go:1071 +0x1c
main.main()
	/go/src/backup_forked/etcd-backup-restore/main.go:29 +0x50
```

It happened because it tries to close the channel which already closed.
https://github.com/gardener/etcd-backup-restore/blob/3ab38affc45a490e45f9aafd6f8734f9a591feee/pkg/server/backuprestoreserver.go#L281


**How to reproduce it (as minimally and precisely as possible)**:
1. Start an `etcd` and backup-restore as sidecar
2. Kill the `etcd` process and restart it.
3. Again kill the `etcd` process, now backup-restore will panics

**Anything else we need to know?**:

**Environment**:

- etcd-backup-restore version:
- etcd-druid version (if relevant):
- Gardener version (if relevant):
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

--- a/pkg/server/backuprestoreserver.go
+++ b/pkg/server/backuprestoreserver.go
@@ -236,6 +236,7 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 				}
 
 				if b.config.SecondarySnapstoreConfig.BackupSyncEnabled {
+					backupGcStop = make(chan struct{})
 					b.logger.Infof("Starting periodic backup copier..")
 					secondary, err := snapstore.GetSnapstore(b.config.SecondarySnapstoreConfig.StoreConfig)
 					if err != nil {
@@ -274,7 +275,9 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 		OnStoppedLeading: func() {
 			if runServerWithSnapshotter {
 				b.logger.Info("backup-restore stops leading...")
-				close(backupGcStop)
+				if b.config.SecondarySnapstoreConfig.BackupSyncEnabled {
+					close(backupGcStop)
+				}
 				handler.SetSnapshotterToNil()
 
 				// TODO @ishan16696: For Multi-node etcd HTTP status need to be set to `StatusServiceUnavailable` only when backup-restore is in "StateUnknown".

